ansible_runner/config/_base.py

```diff
@@ -187,6 +193,7 @@ def _prepare_env(self, runner_mode='pexpect'):
         self.container_image = self.settings.get('container_image', self.container_image)
         self.container_volume_mounts = self.settings.get('container_volume_mounts', self.container_volume_mounts)
         self.container_options = self.settings.get('container_options', self.container_options)
+        self.container_auth_data = self.settings.get('container_auth_data', self.container_auth_data)
```

Comment:
You have not added code for handling the option in the CLI (`ansible-runner/__main__.py`).

Reply:
This feature does not make any sense for `ansible-runner run`, because if you had the relevant host / username / password you should instead use a command like `podman login` to set up the config in the user-specific proper location.

However, it is completely unreasonable to login if you are running remotely.

If anything, this should be CLI options attached to `ansible-runner transmit`, because this is where AWX needs it, and it is the only use case where it makes sense to transmit the auth for a container registry. The need for this is created by 2 mutual requirements of (1) providing container registry credentials and (2) not exposing those credentials to the job runtime environment.

For now, I certainly think it is sufficient to leave this as a feature of the python interface.

Reply:
@Shrews just wanted to check in again on this. Is this still a concern?

Reply:
It seems like we should aim to keep parody between the Python interface and the CLI. I suppose if we have to deviate anywhere, first implementing a new feature in the Python interface before exposing it on the CLI is what I would prefer.

Reply:
It is possible via the CLI by defining `container_auth_data` in the `env/settings` file. We have been manually testing this use case.